### ISSUE TITLE
Have admin credentials in separate location in BOSH-Lite stubs

### ISSUE THREAD

**Issue by Amit-PivotalLabs** [cloudfoundry/cf-release#874]
> Original asker: @ljfranklin 
> Original location: https://cloudfoundry.slack.com/archives/release-integration/p1451438082001200 (Slack)

----

hey gang, would you guys be opposed to a PR that adds the following properties to the cf-release bosh-lite stub:

```
properties:
  admin_username: admin
  admin_password: admin
  skip_ssl_validation: true
```

We're trying to clean up the stubs for the mysql service, and our errands (e.g. broker-registrar) need creds to talk to CF. It would be nice to pull these creds from a CF manifest rather than asking the operator to copy-paste them into one of our stubs. Unfortunately, we didn't see an easy way to get these creds from the current manifest properties without parsing the scim string (`username|password|blah|blah`).

**Comment by Amit-PivotalLabs** [comment]
Hey @ljfranklin 

(I moved the question over here to GitHub, so it doesn't get buried in Slack.)

We definitely need a better way of allowing components to share certain credentials.  Few questions...

1. Do you only need this for BOSH-Lite or would you need this for the stubs for every IaaS?
1. Exactly what permissions are required for the stuff you want to do, would it work to just have a separate user for the specific purpose (and with the specific permissions) needed for what you’re trying to do?
1. Even then, though, it’s hard to parse the credentials out, would be nice if the user list were in a saner format, i’ll try to see what the UAA team thinks about that /cc @mbhave @fhanik @sreetummidi 

I'd like to think about the right way to do this, and how this would work in the world of BOSH links /cc @cppforlife @mariash

**Comment by ljfranklin** [comment]
@Amit-PivotalLabs Thanks for creating this issue for us.

> Do you only need this for BOSH-Lite or would you need this for the stubs for every IaaS?

Thinking more generally, we would need this for every IaaS for a consistent experience.

> Exactly what permissions are required for the stuff you want to do...

Looking at the [CF Services doc](https://docs.cloudfoundry.org/services/access-control.html), the commands we need to run (e.g. `enable-service-access`) must be run by an administrator. Looking quickly at the [UAA docs](https://docs.cloudfoundry.org/adminguide/uaa-user-management.html), I'm guessing we need an admin user with the `cloud_controller.admin` authority, but would need to confirm.

> Even then, though, it’s hard to parse the credentials out...

Agreed. We're not opposed to doing a bit of parsing, but it would be nice to have a cleaner format that services could easily hook into.

**Comment by Amit-PivotalLabs** [comment]
@ljfranklin I spoke with @sreetummidi and she thinks it's a good idea for `users` to an array of both the current pipe-separated strings as well as a more structured thing, e.g.

```
properties:
  uaa:
    scim:
      users:
        - oldstyle|p@ssw0rd|some.scope,some.other.scope
        - name: newuser
          password: pteradactyl
          scopes:
            - scope.one
            - scope.two
```

The `name` key will be clutch, as it will allow you to reference it in spiff.  I'm not sure the priority of said story, but I'd rather wait for something like that to land before shoving otherwise unnecessary stuff into the stubs.

Let me know if you want to discuss further, or if you think it's cool to close out this issue.

**Comment by Amit-PivotalLabs** [comment]
By the way, here is that story: https://www.pivotaltracker.com/story/show/111089660

**Comment by Amit-PivotalLabs** [comment]
@ljfranklin I'll close this out, feel free to reach out if you think it still needs discussion.

**Comment by ljfranklin** [comment]
Thanks for running with this @Amit-PivotalLabs! We'll follow up with @sreetummidi via Tracker.

**Comment by fhanik** [comment]
https://github.com/cloudfoundry/uaa-release/commit/bd8c61406c60e79364f170a1fe22154f8f0551d0
